src/NHSD.BuyingCatalogue.Solutions.Application/Queries/GetPricingBySolutionId/GetPriceByCatalogueItemIdHandler.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using AutoMapper;
+using MediatR;
+using NHSD.BuyingCatalogue.Solutions.Application.Domain.Pricing;
+using NHSD.BuyingCatalogue.Solutions.Application.Persistence;
+using NHSD.BuyingCatalogue.Solutions.Contracts.Pricing;
+using NHSD.BuyingCatalogue.Solutions.Contracts.Queries;
+
+namespace NHSD.BuyingCatalogue.Solutions.Application.Queries.GetPricingBySolutionId
+{
+    internal sealed class GetPriceByCatalogueItemIdHandler : IRequestHandler<GetPriceByCatalogueItemIdQuery, IEnumerable<ICataloguePrice>>
+    {
+        private readonly PriceReader _priceReader;
+        private readonly CatalogueItemVerifier _verifier;
+        private readonly IMapper _mapper;
+
+        public GetPriceByCatalogueItemIdHandler(PriceReader priceReader, CatalogueItemVerifier verifier, IMapper mapper)
+        {
+            _priceReader = priceReader;
+            _verifier = verifier;
+            _mapper = mapper;
+        }
+
+        public async Task<IEnumerable<ICataloguePrice>> Handle(GetPriceByCatalogueItemIdQuery request, CancellationToken cancellationToken)
+        {
+            await _verifier.ThrowWhenMissingAsync(request.CatalogueItemId, cancellationToken);
```

Comment:
I am thinking we don't need the `_verifier` as the request is trying to filter on a catalogue item ID, if that catalogue item ID doesn't exist then in the context of this request there are no prices for it. So I think it should return an empty list & we shouldn't bother with the not found status. This would mean letting the request go all the way to the database & wouldn't need a second call to the DB. What do you think?

Reply:
Totally agree, I put it in as I misunderstood the ticket and thought this endpoint is to replace the previous one which returns prices by Solution Id. Given the clarification, the verifier doesn't make sense, I'll get rid of it 👍